joomla/joomla-cms
# Refactor 2FA from FOF to Joomla Core

#### Summary of Changes
- Refactored 2FA from FOF to Joomla Core
- Removed dependecies to FOF
- ~~Refactored 2FA templates to JLayouts~~ Reverted back for B/C, removed the FOF part that resolves the template paths and used JPluginHelper instead.
#### Testing Instructions
- **Step 1** Activate the Twofactor Auth plugins ( Google Auth, Yubikey) in the backend 
- **Step 2** Frontend: Go to the "Edit user profile" page and activate 2FA (Google Auth) and set up the authenticator
- **Step 2.1** Select Yubikey from the dropdown menu and it's template should be loaded ( Yubikey FOF dependecy was only related to the template)
- **Step 3** Logout and log in using your Google authenticator ( Frontend & Backend)
- **Step 4** After applying the patch Step 1-3 should work as before.

To test that custom templates still work create a "form.php" in "templates/protostar//html/plg_twofactorauth_totp" (Google Auth) with an echo() or similar and 
go to Frontend -> Edit User profile , select Google Autheticator and your form's content should be shown.
